boot/zephyr/Makefile

```diff
@@ -1,4 +1,4 @@
-subdir-ccflags-y += -DMCUBOOT_TARGET_CONFIG='"$(BOARD).h"'
+subdir-ccflags-y += -DMCUBOOT_TARGET_CONFIG='"$(BOARD).inc"'
```

Comment:
Changing the extension of these files will break numerous tools/editors that use the extension to determine the file type.

I would suggest something along the lines of:
* Move target.h into boot/zephyr/include
* Remove boot/zephyr/targets from the include path entirely
* Perhaps move it to boot/zephyr/include/targets
* Have the target files include target/name.h instead of just name.h

If you want to really protect against direct inclusion, they could #error if there is a define not defined, and then define that in target.h before doing the include, but this might be overkill.